src/BeneficiariesFile.hs

```diff
@@ -39,15 +45,18 @@ parseBeneficiary conf = toBeneficiary . words
   where
     toBeneficiary :: [Text] -> Either Text Beneficiary
     toBeneficiary [addr, amt, ac] =
-      makeBeneficiary addr (parseAmt amt) (parseAsset ac)
-    -- Second arg could be amount or asset class, so we try to parse as asset first, if not, then amount
+      makeBeneficiary
+        addr
+        (withMaybeOverride (parseAmt amt) conf.dropAmount)
+        (withMaybeOverride (parseAsset ac) conf.assetClass)
+    -- Second arg could be amount or asset class, so we try to parse as an amount first, if not, then asset
     -- Then fill in the Beneficiary with the data we have left, failing if we're missing anything
     toBeneficiary [addr, assetOrAmt] = do
       eAssetOrAmt <- parseAssetOrAmt assetOrAmt
       makeBeneficiary
         addr
-        (maybeToMissing "quantity" $ rightToMaybe eAssetOrAmt <|> conf.dropAmount)
-        (maybeToMissing "assetclass" $ leftToMaybe eAssetOrAmt <|> conf.assetClass)
+        (maybeToMissing "quantity" $ conf.dropAmount <|> rightToMaybe eAssetOrAmt)
+        (maybeToMissing "assetclass" $ conf.assetClass <|> leftToMaybe eAssetOrAmt)
```

Comment:
Was a bug here where the defaults weren't being used if it parsed correctly, this was the fix (along with `withMaybeOverride`)

Reply:
The defaults shouldn't be used if there is a value provided, the idea is that the defaults are used only if the beneficiaries doesn't provide a value.
With this change, if we have `--amount` set and provide a value to a specific beneficiary, would the `--amount` be used?

Reply:
According to the documentation, the flag is supposed to override the file:

> drop-amount: Amount of tokens to send to each beneficiary (overrides beneficiaries file config)

I agree it's better to be default, rather than override

Reply:
(I realise I used the word default earlier which may have added confusion)

Reply:
Oh! sorry I misread the docs, yeah they're wrong. Its supposed to be a default